### PR TITLE
Some additions to `Data.List.Relation`

### PR DESCRIPTION
**Pull request by MatthewDaggitt**
In line with trying to get all major changes to `Data.X.Relation` modules out of the way before releasing 0.15:
- Added new modules `Data.List.Relation.Equality.Setoid/DecSetoid/Propositional/DecPropositional`. I've prefixed the proofs in these modules with the name of the equality itself (e.g. `≋-refl` and not just `refl`), as if you are reasoning about list equalities, you've almost certainly opened the original `Setoid` as well. This therefore avoids any resulting name clashes. If people want the "plain" names, they can always just open `≋-setoid`.
- Deprecated the name `Rel` in favour of `Pointwise` in the module `Data.List.Relation.Pointwise`. This brings it into line with `Data.Vec.Relation.InductivePointwise`. It also gets rid of the all the horrible conflicts with `Rel` from `Relation.Binary` which I've always found tedious. (Notice how everywhere where it was used in the standard library, it was always prefixed or renamed first).
- Added a few extra properties about `++`, `tabulate` and `concat` (one of the original reasons why relations were moved to `Data.X` in the first place).

Any feedback on the design of the `Equality` modules?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -254,11 +258,14 @@ Backwards compatible changes
 
 * Added support for GHC 8.2.2.
 
-* New modules `Data.Table`, `Data.Table.Base`,
-  `Data.Table.Relation.Equality` and `Data.Table.Properties`. A `Table` is a
-  fixed-length collection of objects similar to a `Vec` from `Data.Vec`, but
-  implemented as a function `Fin n → A`. This prioritises ease of lookup as opposed
-  to `Vec` which prioritises the ease of adding and removing elements.
+* New module `Data.Word` for new builtin type `Agda.Builtin.Word.Word64`.
```
Comment (gallais) [inline CHANGELOG.md]: You probably didn't mean to commit this

Comment (MatthewDaggitt) [reply CHANGELOG.md]: Thanks, nice spot! Didn't branch off the right place, and clearly mishandled the merge.

**Comment by MatthewDaggitt** [comment]
Note to self. Need to uncomment the deprecated `Rel` names before committing.

**Comment by MatthewDaggitt** [comment]
And add `isDecEquivalence` and `decSetoid` to `DecidableEquality`

**Pull request merged**